### PR TITLE
Users authenication

### PR DESCRIPTION
**Pull request by Michaeljogoh**
Update user-authentication branch

### CODE REVIEW

File: controllers/authController.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+
+
+
+// Sign Up
+const registerUsers = async (req , res)  =>{
+    const {name , email ,  password , password2 } = req.body;
+    //Validation
+    if(!name || !email || !password || !password2 ){
+        res.status(204).json({error:"Please fill in all field"})
```
Comment (Asylcreek) [inline controllers/authController.js]: Any status code less than 400 is a success to the browser.

You are supposed to be using `400 (bad request)`

File: controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    if(password < 6){
+       res.status(204).json({error:"password must not be less than six characters"})
+    }
+    
+    await  Users.findOne({email : email})
+    if(email){
+        res.status(204).json({error:"Email already registered"})
+     } else {
```
Comment (Asylcreek) [inline controllers/authController.js]:
```jsx
const user = await Users.findOne({email})
if(user){
...
```

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: controllers/authController.js (hunk 1 above)
Comment (Michaeljogoh) [reply controllers/authController.js]: Ok , i will change it

File: controllers/authController.js (hunk 2 above)
Comment (Michaeljogoh) [reply controllers/authController.js]: ok sir

**Review by Asylcreek — APPROVED** [review]

**Pull request merged**